addons/mail/static/src/models/partner/partner.js

```diff
@@ -263,7 +266,11 @@ function factory(dependencies) {
         }),
         country: many2one('mail.country'),
         display_name: attr({
+            compute: '_computeDisplayName',
             default: "",
+            dependencies: [
+                'user'
+            ],
```

Comment:
dependency should be `display_name` and `userDisplayName` instead (those are the fields that are actually used to compute)